### DISCUSSION TITLE
Unable to connect to a new WiFi/Ethernet connection without letting the Network Manager editting the resolv.conf file

### DISCUSSION THREAD

**Discussion by PrimAvalou** [DNSCrypt/dnscrypt-proxy#1769] [Q&A]
Hi everyone !

Firstly I am a Linux beginner (Manjaro distribution / XFCE desktop environment), so sorry if I just didn't found the correct tutorial to make it work. I only follow instructions that I understand what they do on my computer, so maybe I went through instructions to prevent this issue without understanding what they really do (but I don't think so).

To install dnscrypt-proxy I followed the following tutorials : 
https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Installation-on-Arch-Linux
https://wiki.archlinux.org/title/Dnscrypt-proxy

Then, dnscrypt-proxy worked (and still works) perfectly. My issue comes when I try to connect my computer to a new connection (for example if I try to connect to a hotel WiFi). As well as I locked the resolv.conf file, the Network Manager can't modify it to enable the connection. What I have to do each time I want to connect to a new connection is to unlock the resolv.conf file, then reboot my computer (so now the Network Manager can modify the resolv.conf file according to the new connection), and finally I replace the resolv.conf file with a backup containing the following lines : 
`nameserver 127.0.0.1`
`options edns0`

I started to look for bash scripts to unlock the resolv.conf file on my computer logout, and replace the resolv.conf once the computer is connected. Or something like that. But I guess there is a proper way to make dnscrypt-proxy automatically work on a new connection. What do you guys advice to make it work at the connection of a new WiFi or Ethernet network ? 

Thanks a lot for your time and answers !

**Comment by jumpsmm7** [comment]
What listen address are you using for your Dnscrypt proxy setup? You could just install something simple like PiHole. Allow Pihole to pick up the Dnscrypt proxy request via it's listen address and handle the rest on its own. (you don't have to use too many block list if you dont want to.)

**Reply by PrimAvalou** [reply]
Listen address is 127.0.0.1 (port 53000) : 
`listen_adresses = ['127.0.0.1:53000', '[::1]:53000']`

PiHole could be interesting indeed. But I would prefer to firstly consider solutions that uses dnscrypt-proxy features, or bash scripts in the case there is no any other way. I mean I ask for solutions to avoid makeshift job from mine (I remind that I am a beginner :) ). If there is no any dnscrypt-proxy feature, I would like to know clean ways using bash scripts. If that's not possible using bash scripts I would consider installing extra-tools (like PiHole).

Thanks for your answer :)
